src/outlier_hub/datasets/ham10k/factory.py

```diff
@@ -1,75 +1,152 @@
+                                                    filename = url_dict['file_name'])
 
     def _prepare(self, split: str):
-        # TODO like this it does not support train and test splits
         preprocessor = HAMPreprocessor(self.storage_connector)
+        raw_samples_identifier = self._get_resource_id(data_type = 'raw', data_split = split + '/raw_images.zip')
+        raw_targets_identifier = self._get_resource_id(data_type = 'raw', data_split = split + '/train_labels.zip')
 
-        samples_identifier = self._get_resource_id(element=split + "/images")
-        targets_identifier = self._get_resource_id(element=split + "/metadata/HAM10000_metadata_sorted.csv")
-        dataset_identifier = self._get_resource_id(element="ham10k.hdf5")
+        preprocessed_dataset_identifier = self._get_resource_id(data_type='preprocessed', data_split='train/' + self.dataset_name)
 
-        preprocessor.preprocess(dataset_identifier=dataset_identifier,
-                                samples_identifier=samples_identifier,
-                                targets_identifier=targets_identifier)
+        preprocessor.preprocess(raw_samples_identifier = raw_samples_identifier,
+                                raw_targets_identifier = raw_targets_identifier,
+                                prep_dataset_identifier = preprocessed_dataset_identifier)
 
     def _get_iterator(self, split: str):
-        dataset_identifier = "ham10k.hdf5"
-        dataset_resource = self.storage_connector.get_resource(identifier=dataset_identifier)
+        if not (self.check_exists()):
+            print('Downloading and preparing the dataset.')
+            if split not in self.splits:
+                raise logging.debug(f"Split {split} is not defined.")
+            self._retrieve_raw()
+            self._prepare(split)
+        dataset_identifier = self._get_resource_id(data_type = 'preprocessed', data_split = os.path.join(split, self.dataset_name))
+        print('dataset_identifier:', dataset_identifier)
+        #dataset_resource = self.storage_connector.get_resource(identifier = dataset_identifier)
         meta = MetaFactory.get_iterator_meta(sample_pos=0, target_pos=1, tag_pos=2)
-        return HAMIterator(dataset_resource, split), meta
+        return HAMIterator(dataset_identifier), meta
+                            
 
     def get_dataset_iterator(self, config: Dict[str, Any] = None) -> Tuple[DatasetIteratorIF, IteratorMeta]:
-        if not (self.check_exists()):
-            self._prepare(split="raw")
         return self._get_iterator(**config)
 
-
-# Code for testing the dataset
 if __name__ == "__main__":
+    
     # get root workind directory path
-    root_path = pathlib.Path.cwd().parents[3]
+    root_path = pathlib.Path.cwd()
+    logger.debug(f' pathlib.Path.cwd(): { pathlib.Path.cwd()}')
 
     # complete path to manual added data
     data_path = os.path.join(root_path, "src/outlier_hub/datasets/ham10k/data")
```

Comment:
Can you specify another path here, please? src is just meant for sourcecode

Reply:
Of course - Done